cypress/integration/images.spec.js

```diff
@@ -45,50 +55,81 @@ const clickOnImageAction = (actionIndex, callback) => {
 		.should('have.class', 'action-item--open')
 
 	// get the popover ID to be able to find the related DOM element
-	cy.get('#viewer .action-item.submenu > div.v-popover > .trigger')
+	return cy.get('#viewer .action-item.submenu > div.v-popover > .trigger')
 		.should('have.attr', 'aria-describedby')
-			.should('contain', 'popover_')
-			.then((popoverId) => {
-				cy.log('Click on the action entry')
-				cy.get('div#' + popoverId)
-					.should('have.class', 'open')
-					.find('li:nth-child(' + actionIndex + ')').click()
+		.should('contain', 'popover_')
+		.then((popoverId) => {
+			cy.log('Click on the action entry')
+			cy.get('div#' + popoverId)
+				.should('have.class', 'open')
+				.find('li:nth-child(' + actionIndex + ')').click()
 				// our job here is done
-				callback(popoverId)
-			})
+			callback(popoverId)
+		})
 }
 
 /**
  * Check if an image is visible in the document
  *
- * @param documentId file ID of the current document
- * @param imageName file name to be checked
+ * @param {number} documentId file ID of the current document
+ * @param {string} imageName file name to be checked
+ * @param {number} imageId file id
+ * @param {number|undefined} index index of image in the document
  */
-const checkImage = (documentId, imageName, imageId) => {
-	cy.log('Check the image is visible and well formed')
-	cy.get('#editor .ProseMirror div.image[data-src="text://image?imageFileName=' + imageName + '"]')
-		.should('be.visible')
-		.find('img')
-			.should('have.attr', 'src')
-				.should('contain', 'apps/text/image?documentId=' + documentId)
-				.should('contain', 'imageFileName=' + imageName)
-	// keep track that we have created this image in the attachment dir
-	attachmentFileNameToId[imageName] = imageId
+const checkImage = (documentId, imageName, imageId, index) => {
+	const encodedName = fixedEncodeURIComponent(imageName)
+
+	cy.log('Check the image is visible and well formed', { documentId, imageName, imageId, index, encodedName })
+	return new Cypress.Promise((resolve, reject) => {
+		cy.get('#editor [data-component="image-view"]')
+			.filter('[data-src="text://image?imageFileName=' + encodedName + '"]')
+			.find('.image__view') // wait for load finish
+			.within(($el) => {
+				// keep track that we have created this image in the attachment dir
+				if (!attachmentFileNameToId[documentId]) {
+					attachmentFileNameToId[documentId] = {}
+				}
+
+				attachmentFileNameToId[documentId][imageName] = imageId
+
+				if (index > 0) {
+					expect(imageName).include(`(${index + 1})`)
+				}
+
+				cy.wrap($el)
+					.should('be.visible')
+					.find('img')
+					.should('have.attr', 'src')
+					.should('contain', 'apps/text/image?documentId=' + documentId)
+					.should('contain', 'imageFileName=' + encodeURIComponent(imageName))
```

Comment:
@eneiluj I don't know exactly the reason for this behavior, but looks like the element appears different when we read the attribute.